WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/WooShipping Create Shipping Labels/WooShipping Split Shipments/WooShippingSplitShipmentsViewModel.swift

```diff
@@ -92,13 +91,62 @@ final class WooShippingSplitShipmentsViewModel: ObservableObject {
         instructions = nil
         dismissedInstructions = true
     }
+
+    func moveSelectedItems(to destination: MoveToShipmentNoticeViewModel.Destination) {
+        moveToNoticeViewModel = nil
+        instructions = nil
+
+        // Step 1: Split items
+        var newShipment = Shipment()
+        var movedItems = [CollapsibleShipmentItemCardViewModel]()
+        for item in currentShipment {
+            let initialQuantity = item.packageItem.quantity.intValue
+            let selectedQuantity = item.numberOfSelectedItems
+            let remainingQuantity = initialQuantity - selectedQuantity
+            if remainingQuantity == 0 {
+                movedItems.append(
+                    CollapsibleShipmentItemCardViewModel(item: item.packageItem, currency: self.order.currency)
+                )
+            } else if selectedQuantity > 0 {
+                let newItem = ShippingLabelPackageItem(copy: item.packageItem, quantity: Decimal(remainingQuantity))
+                newShipment.append(
+                    CollapsibleShipmentItemCardViewModel(item: newItem, currency: self.order.currency)
+                )
+                let movedItem = ShippingLabelPackageItem(copy: item.packageItem, quantity: Decimal(selectedQuantity))
+                movedItems.append(
+                    CollapsibleShipmentItemCardViewModel(item: movedItem, currency: self.order.currency)
+                )
+            } else if selectedQuantity == 0 {
+                newShipment.append(
+                    CollapsibleShipmentItemCardViewModel(item: item.packageItem, currency: self.order.currency)
+                )
+            }
+        }
+
+        // Step 2: Update the current shipment
+        let currentIndex = selectedShipmentIndex ?? 0
+        shipments[currentIndex] = newShipment
+        configureSelectionCallback()
+        configureSectionHeader()
+
+        // Step 3: Add new or update existing shipment
+        switch destination {
+        case .newShipment:
+            shipments.append(movedItems)
+
+        case .existingShipment(let index):
```

Comment:
I think we should check for existing items and merge the new items into existing collapsible card. Right now, we are creating more than one collapsible card for the same product. WDYT? 

https://github.com/user-attachments/assets/ed284fc8-0a10-4e80-bf64-73f342883801

Reply:
Good catch! I updated the logic in 7e5d3076bfba293f299586710351e3e7c8545776 and added a test for that in b20e8964fce342f4c96d5bf10212aa55cb7b7831.